src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -1583,8 +1583,14 @@ static int xocl_cma_mem_alloc(struct xocl_dev *xdev, uint64_t size)
 		return -EINVAL;
 	}
 
+/* TODO: check that MAX_PAGE_ORDER is actually well used and this code is not
+ * too pessimistic */
+#if KERNEL_VERSION(6, 8, 0) <= LINUX_VERSION_CODE
+	if (page_sz > (PAGE_SIZE << (MAX_PAGE_ORDER-1))) {
+#else
 	if (page_sz > (PAGE_SIZE << (MAX_ORDER-1))) {
-		DRM_WARN("Unable to allocate with page size 0x%llx", page_sz);
+#endif
+	DRM_WARN("Unable to allocate with page size 0x%llx", page_sz);
```

Comment:
indent?

Reply:
Oh. Thanks for mention this. NR_PAGE_ORDERS should be used instead of MAX_PAGE_ORDER.

Reply:
I put some links to some discussion in each commit message for reference.
So, after rereading the literature and some changes like https://lwn.net/ml/linux-kernel/20231228144704.14033-2-kirill.shutemov@linux.intel.com/
I think the XRT code was not pessimistic about the allocation (and at least avoided the floppy-disk bughttps://lwn.net/ml/linux-mm/20230315113133.11326-4-kirill.shutemov@linux.intel.com/  ;-) )
So now the modern way to write it is just
```c
if (page_sz > (PAGE_SIZE << MAX_ORDER_NR_PAGES)) {
```
by using definitions from https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/include/linux/mmzone.h?id=e8f897f4afef0031fe618a8e94127a0934896aba#n34

But actually this is the definitive answer that seems to have been used for quite some Linux releases, as it it is in 5.4 and 4.18.